### PR TITLE
348 Separate instructions for users and developers

### LINKED ISSUE #212: Readme has no instructions for non-docker CLI
**Is your feature request related to a problem? Please describe.**
The readme has instructions for setting up hazen using the python virtual environments as an alternative to docker but the usage instructions only make use of the hazen-app docker script.

**Describe the solution you'd like**
We should add usage instructions and examples for using the virtual environment implementation

**Describe alternatives you've considered**
The only alternative is _not_ having these instructions, which will lead to confusion when people set up a venv but then follow the existing instructions and end up running the docker version anyway

### LINKED ISSUE #300: Add information on how to update hazen to readme
**Is your feature request related to a problem? Please describe.**
The readme currently covers the various ways of installing hazen, but i don't think we're explicitly telling people how to upgrade. 

**Describe the solution you'd like**
A description of how to update hazen using pip, source, (docker not relevant here as latest tag image used by default)

### LINKED ISSUE #303: Update contributing guidelines
We briefly discuss Releasing guidelines on the [README](https://github.com/GSTT-CSC/hazen/blob/main/README.md) but I think we should review/update both this and the [Contributing](https://github.com/GSTT-CSC/hazen/blob/main/CONTRIBUTING.md) guidelines (which also has a [Releasing](https://github.com/GSTT-CSC/hazen/blob/main/CONTRIBUTING.md) section).

### LINKED ISSUE #348: separate READme for users and developers
Current READme file presents instructions for both users and developers and different sections apply to either.
I suggest having a separate "users readme" which would be the main one currently available and the developer/contributor specific instructions would be moved to a new "developer READme".

@tomaroberts let me know what you think

### PR DESCRIPTION
**Pull request by sophie22**
Update the existing READme file to only retain instructions relevant for users, move all developer specific instructions and update the existing contributing.md file.

### CODE REVIEW

**Comment by sophie22** [comment]
This will close #348 #303 and resolves #212 
TODO: instructions for how to use pip upgraded and install latest version of hazen #300

**Comment by tomaroberts** [comment]
Tom TBD – what to do with CONTRIBUTING.md. Remove entirely, or update.

**Comment by tomaroberts** [comment]
Going to create new file. Inspiration: [here](https://github.com/Project-MONAI/MONAI/blob/dev/CONTRIBUTING.md) and [here](https://github.com/Project-MONAI/monai-deploy/blob/main/CONTRIBUTING.md)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,38 +1,115 @@
...
+contribution.
+2. If there is no existing Issue, create a new Issue for your contribution:
+   - Select the `Bug report` or `Feature request` template
+   - Fill in the Issue according to the template
+   - Add relevant Labels to the issue: `Enhancement`, `Bug`, `MRI`, `DICOM`, etc
+3. Create a new branch from `main`
+   - Name the branch with the issue number and a short description, e.g.: `123-snr-bugfix`
+4. Make your code changes (see guidance above)
+5. Perform unit tests on your machine: `pytest tests/`
+6. Create a [Pull Request](https://github.com/GSTT-CSC/hazen/pulls)
```
Comment (heyhaleema) [inline CONTRIBUTING.md]:
```suggestion
6. Create a [Pull Request](https://github.com/GSTT-CSC/hazen/pulls) (PR)
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,38 +1,115 @@
...
+   - Describe your changes
+   - Describe why you coded your change this way
+   - Describe any shortcomings or work still to do
+   - Cross-reference any relevant Issues
+7. One of the hazen team members will review your Pull Request and then:
```
Comment (heyhaleema) [inline CONTRIBUTING.md]:
```suggestion
7. One of the hazen team members will review your PR and then:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,38 +1,115 @@
...
+   - Ask questions
+   - Request any changes
+   - Merge into `main` – thank you and congratulations on contributing to hazen!
+
+
+## 3) Release Process
+
+The Release Process involves approving and then merging all Pull Requests identified for the new release of hazen.
```
Comment (heyhaleema) [inline CONTRIBUTING.md]:
```suggestion
The Release Process involves approving and then merging all PRs identified for the new release of hazen. 
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,38 +1,115 @@
...
+   - Create a tag equal to the version number, e.g. 1.2.1
+   - Select `main` as the Target branch from which to create the Release
+   - Select "Generate release notes" to automatically pull in the Merge Commit messages
+   - Amend the Release Notes:
+     - Add any other changes not automatically generated
+     - Edit Release Notes for typos and formatting consistency
+7. Publish Release
+   - This will trigger the [publishing GitHub Actions](https://github.com/GSTT-CSC/hazen/tree/main/.github/workflows) 
+   which push the released version to DockerHub and [PyPI](https://pypi.org/project/hazen/).
+8. Send a Release email to the hazen users!
```
Comment (heyhaleema) [inline CONTRIBUTING.md:115]: Maybe we should include a template or screenshot of previous Release emails?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,38 +1,115 @@
...
+   - Investigate any which are failing
+3. Merge the PR into the `main` branch
+   - Provide a brief description of changes in the PR textbox as this forms the Release Notes 
+
+For a new release:
+4. Close all related Issues resolved by the merged PRs
+5. **Important**: Update version numbers across the repo:
+   - Update version number in `hazenlib/_version.py`
+     - (This is automatically propagated into `docs/source/conf.py`, `hazenlib/__init__.py` and `setup.cfg`)
+   - Update version number in `CITATION.cff`
```
Comment (heyhaleema) [inline CONTRIBUTING.md]: The formatting for these bullet points are little odd:

image

Maybe a newline is needed after "For a new release"?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,38 +1,115 @@
-## How to contribute
+## Table of Contents
+- [Introduction](#introduction)
+- [1) How to make and test code changes](#1-how-to-make-and-test-code-changes)
+- [2) Developer Process for Contributing](#2-developer-process-for-contributing)
+- [3) Release Process](#3-release-process)
```
Comment (heyhaleema) [inline CONTRIBUTING.md]: Is there a way to change these bullet points to a numbered list or maybe as an indented numbered list? For example:

- Introduction
  1. How to make and test code changes
  2. Developer Process for Contributing
  3. Release Process

**Review by heyhaleema — CHANGES_REQUESTED** [review]
Overall, LGTM - great work, @sophie22 and @tomaroberts ✨ I've just added a couple of comments and suggestions, but no major changes requested.

File: CONTRIBUTING.md (hunk 6 above)
Comment (tomaroberts) [reply CONTRIBUTING.md]: How about we just name them 1-4 – agree it's a bit weird the Intro not being numbered. So:

1) Introduction
2) How to make and test code changes
3) Developer Process for Contributing
4) Release Process

?

File: CONTRIBUTING.md (hunk 4 above)
Comment (tomaroberts) [reply CONTRIBUTING.md:115]: Good idea. Will add.

File: CONTRIBUTING.md (hunk 5 above)
Comment (tomaroberts) [reply CONTRIBUTING.md]: Ohh, weird, doesn't look like that in PyCharm editor. I will fix, thanks.

**Comment by tomaroberts** [comment]
@heyhaleema – thanks v much for comments. Changes implemented. If you're happy, please merge :)

**Comment by tomaroberts** [comment]
Ok, now it's ready 😂

**Review by heyhaleema — APPROVED** [review]
LGTM 👍

**Pull request merged**